libpod/options.go

```diff
@@ -518,6 +549,216 @@ func WithExitCommand(exitCommand []string) CtrCreateOption {
+
+		return nil
+	}
+}
+
+// WithMountNSFromPod indicates the the container should join the Mount namespace of
+// its pod
+func WithMountNSFromPod(p *Pod) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+
+		if p == nil {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in was nil. Container may not be associated with a pod")
+		}
+
+		if ctr.config.Pod == "" {
+			return errors.Wrapf(ErrInvalidArg, "container is not a member of any pod")
+		}
+
+		if ctr.config.Pod != p.ID() {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in is not the pod the container is associated with")
+		}
+
+		infraContainer, err := p.InfraContainerID()
+		if err != nil {
+			return err
+		}
+		ctr.config.MountNsCtr = infraContainer
+
+		return nil
+	}
+}
+
+// WithNetNSFromPod indicates the the container should join the network namespace of
+// its pod
+func WithNetNSFromPod(p *Pod) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+
+		if p == nil {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in was nil. Container may not be associated with a pod")
+		}
+
+		if ctr.config.Pod == "" {
+			return errors.Wrapf(ErrInvalidArg, "container is not a member of any pod")
+		}
+
+		if ctr.config.Pod != p.ID() {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in is not the pod the container is associated with")
+		}
+
+		infraContainer, err := p.InfraContainerID()
+		if err != nil {
+			return err
+		}
+		ctr.config.NetNsCtr = infraContainer
+
+		return nil
+	}
+}
+
+// WithPIDNSFromPod indicates the the container should join the PID namespace of
+// its pod
+func WithPIDNSFromPod(p *Pod) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+
+		if p == nil {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in was nil. Container may not be associated with a pod")
+		}
+
+		if ctr.config.Pod == "" {
+			return errors.Wrapf(ErrInvalidArg, "container is not a member of any pod")
+		}
+
+		if ctr.config.Pod != p.ID() {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in is not the pod the container is associated with")
+		}
+
+		infraContainer, err := p.InfraContainerID()
+		if err != nil {
+			return err
+		}
+		ctr.config.PIDNsCtr = infraContainer
+
+		return nil
+	}
+}
+
+// WithUTSNSFromPod indicates the the container should join the UTS namespace of
+// its pod
+func WithUTSNSFromPod(p *Pod) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+
+		if p == nil {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in was nil. Container may not be associated with a pod")
+		}
+
+		if ctr.config.Pod == "" {
+			return errors.Wrapf(ErrInvalidArg, "container is not a member of any pod")
+		}
+
+		if ctr.config.Pod != p.ID() {
+			return errors.Wrapf(ErrInvalidArg, "pod passed in is not the pod the container is associated with")
+		}
```

Comment:
Not a fix for this PR by any means, but it looks like we do these three checks in a number of routines here.  Might be good to have a validateContainerInPod() or some such in a utils class.

Reply:
I'll make the fix after this goes through!

Reply:
rethinking on the naming, might be better to go with "validContainerInPod", or some form of "is..." name with a boolean return.  As you see fit.

Reply:
works for me.